compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -3682,6 +3693,133 @@ private BLangExpression createConditionForMappingBindingPattern(BLangMappingBind
         return statementExpression;
     }
 
+    private BLangExpression createConditionForErrorBindingPattern(BLangErrorBindingPattern errorBindingPattern,
+                                                                  BLangSimpleVarRef matchExprVarRef) {
+        BType bindingPatternType = errorBindingPattern.type;
+        Location pos = errorBindingPattern.pos;
+
+        BLangSimpleVariableDef resultVarDef = createVarDef("errorBindingPatternResult$", symTable.booleanType, null,
```

Comment:
Same comment as the previous PR regarding using a counter and moving to `BLangAnonymousModelHelper`.